File: Java/BasicSample/app/src/main/java/com/neyogiry/android/sample/DatePresenter.java
package com.neyogiry.android.sample;

import java.util.ArrayList;
import java.util.Calendar;

public class DatePresenter implements DateInterfaces.Model {

    private DateInterfaces.View mView;
    private DateModel mModel;

    public DatePresenter(DateInterfaces.View view) {
        this.mView = view;
        this.mModel = new DateModel();
    }

    public void addDate() {
        String time = getTimeCurrent();
        mModel.saveDate(time, this);
    }

    private String getTimeCurrent() {
        String timeCurrent;
        Calendar calendar = Calendar.getInstance();
        timeCurrent = calendar.getTime().toString();
        return timeCurrent;
    }

    @Override
    public void onResponse(ArrayList<String> dates) {
        mView.showDateList(dates);
    }

    @Override
    public void onError(String error) {
        mView.showError(error);
    }
}

File: Kotlin/BasicSample/app/src/main/java/com/neyogiry/android/sample/DatePresenter.kt
package com.neyogiry.android.sample

import java.util.*
import kotlin.collections.ArrayList


class DatePresenter(onView: DateInterfaces.View): DateInterfaces.Model {

    private var mView: DateInterfaces.View
    private var mModel: DateModel

    init {
        mView = onView
        mModel = DateModel()
    }

    fun addDate() {
        val time = getTimeCurrent()
        mModel.saveDate(time, this)
    }

    private fun getTimeCurrent(): String {
        val timeCurrent: String
        val calendar = Calendar.getInstance()
        timeCurrent = calendar.time.toString()
        return timeCurrent
    }

    override fun onResponse(any: Any) {
        mView.showDateList(any as ArrayList<String>)
    }

    override fun onError(error: String) {
        mView.showError(error)
    }


}